### DISCUSSION TITLE
Milvus Standalone Mode

### DISCUSSION THREAD

**Discussion by nairan-deshaw** [milvus-io/milvus#38081] [Q&A and General discussion]
Hi, we're exploring the standalone mode for running multiple small scale applications ( less that 100M vectors at max). We came across embedded and non embedded mode for this deployment. Can someone suggest the differences and when to prefer either of the two? In  addition what are the storage projections with respect to the data? The milvus sizing tool seems to be built for cluster mode, we would like to get an estimate for Milvus standalone mode as well.

TIA!

**Comment by yhmo** [comment]
Embed mode: https://milvus.io/docs/install_standalone-docker.md
Non-embed mode: https://milvus.io/docs/install_standalone-docker-compose.md

Embed mode: there is only one container, the etcd is embeded in milvus main process, data is persisted inside the container. 
The data paths inside the container are mapped to local folder "volumes":
![Screenshot from 2024-11-29 11-23-07](https://github.com/user-attachments/assets/96a1166d-1807-4aca-afc2-75db00977eeb)


Non-embed mode: there are 3 containers: etcd, minio and milvus, data is persisted inside the minio container.
The data paths inside the 3 containers are mapped to local folder "volumes":
![Screenshot from 2024-11-29 11-21-54](https://github.com/user-attachments/assets/193c173f-079d-4a56-a71c-1f935b051253)


For standalone mode, typically the required disk space is about 2x ~ 3x of the raw data size. The required memory capacity is not a static value. Since all the index node, query node, data node are in one process, when you insert data, the index node builds index, the data node compacts segments and flushes data into disk, the query node loads new data into memory, the peak of memory usage might be high.

**Comment by nairan-deshaw** [comment]
Thanks @yhmo!

Could you also please throw some light on when to use embedded vs non embedded mode and what are the tradeoffs in using one over the other?

**Reply by yhmo** [reply]
To the upper applications, there is no much difference.
One container for embedded mode, and 3 containers for embedded mode. Although the number of containers is different, all the data/metadata is stored on local disk. and both can be started/stopped by one command.

**Reply by nairan-deshaw** [reply]
Thanks!
